src/main/scala/org/constellation/domain/redownload/MajorityStateChooser.scala

```diff
@@ -30,35 +30,40 @@ class MajorityStateChooser(id: Id) {
     proposersCount: Int
   ): Option[ExtendedSnapshotProposal] =
     proposals.toList.sortBy(o => (-o.trust, -o.percentage, o.hash)).headOption.flatMap { s =>
-      if (areAllProposals(proposals, proposersCount) || isClearMajority(proposals)) {
-        s.some
-      } else None
+      if (areAllProposals(proposals, proposersCount)) s.some
+      else None
     }
 
   private def areAllProposals(proposals: Set[ExtendedSnapshotProposal], proposersCount: Int): Boolean =
     proposals.toList.map(_.n).sum == proposersCount
 
-  private def isClearMajority(proposals: Set[ExtendedSnapshotProposal]): Boolean =
-    proposals.toList.count(_.totalPercentage >= 0.5) == 1
+  private def roundError(value: Double) =
+    if (value < 1.0E-10 && value > -1.0E-10) 0d
+    else value
 
   private def mergeByHeights(
     proposals: Map[Id, SnapshotProposalsAtHeight],
     proposersCount: Int
   ): Map[Long, Set[ExtendedSnapshotProposal]] =
     proposals.map {
-      case (id, v) =>
-        v.mapValues(p => ExtendedSnapshotProposal(p.hash, p.reputation.getOrElse(id, 0.0), Set(id), 1, proposersCount))
+      case (id, v: Map[Long, SnapshotProposal]) =>
+        v.map{ case (height, p) =>
+          val totalReputation = proposals.values.flatMap { heightSnapshotProposalMap: Map[Long, SnapshotProposal] =>
+            heightSnapshotProposalMap.get(height).map(_.reputation.getOrElse(id, 0d))
```

Comment:
It takes all peers' reputation, not only the proposers of a particular proposal, as you wrote on slack.